chat-android/src/test/java/com/ably/chat/AtomicCoroutineScopeTest.kt

```diff
@@ -0,0 +1,287 @@
+package com.ably.chat
+
+import io.ably.lib.types.AblyException
+import io.ably.lib.types.ErrorInfo
+import java.util.concurrent.LinkedBlockingQueue
+import kotlin.time.DurationUnit
+import kotlin.time.toDuration
+import kotlinx.coroutines.CompletableDeferred
+import kotlinx.coroutines.CoroutineName
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.awaitAll
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.test.runTest
+import kotlinx.coroutines.withContext
+import org.hamcrest.CoreMatchers.containsString
+import org.junit.Assert
+import org.junit.Test
+
+class AtomicCoroutineScopeTest {
+
+    @Test
+    fun `should perform given operation`() = runTest {
+        val atomicCoroutineScope = AtomicCoroutineScope()
+        val deferredResult = atomicCoroutineScope.async {
+            delay(3000)
+            return@async "Operation Success!"
+        }
+        Assert.assertFalse(atomicCoroutineScope.finishedProcessing)
+        val result = deferredResult.await()
+        assertWaiter { atomicCoroutineScope.finishedProcessing }
```

Comment:
This helper method is a part of `chat-android/src/test/java/com/ably/chat/TestUtils.kt`, since it's in the same package, we don't need to import it right?